src/components/DeleteComment.jsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState } from "react";
+import { getCommentsByArticleId, deleteComment } from "../api";
+
+ export function DeleteComment({comment, setComments, setClickDelete}) {
+
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const [message, setMessage] = useState("");
+
+  const handleDelete = (e) => {
+    e.preventDefault();
+    if(username !== comment.author) {
+        setMessage("Username invalid. Check spelling and ensure you are only trying to delete your own comment!");
+        return;
+    }
+    if(password !== "password") {
+        setMessage("Incorrect password.");
+        return;
+    }
+        deleteComment(comment.comment_id)
+          .then(() => {
+            setComments((currentcomments) => {
+             return currentcomments.filter((item) => item.comment_id !== comment.comment_id )})
+            setMessage("Comment deleted successfully.");
+            setClickDelete(false)
+          })
+          .catch((error) => {
+            console.log(error)
+            setMessage("Error deleting comment:"+ error);
+          });
+
+  };
+
+  return (
+    <div>
+      <h3>Delete Comment</h3>
+      <form onSubmit={handleDelete}>
+        <div>
+          <label htmlFor="username">Username:</label>
+          <input
+            type="text"
+            value={username}
+            onChange={(e) => setUsername(e.target.value)}
+            required
+          />
```

Comment:
I'm not sure what the point of typing the username in to delete a comment is?
Instead of this, it would be much better to either have a login page where you can choose your current user, or have a logged user that's hardcoded.

Either way - we'd only want to be able to delete the comments of the user that we are logged in as. Comments by other users shouldn't have a delete button